src/components/Agents/Agents.vue

```diff
@@ -106,31 +108,53 @@ export default {
     }
   },
   methods: {
+    ...mapActions('alert', ['setAlert']),
     async clearUnhealthyAgents() {
-      this.agents
-        .filter(
-          agent => agent.secondsSinceLastQuery > 60 * this.unhealthyThreshold
-        )
-        .forEach(agent => {
-          agent.isDeleting = true
-
-          this.$apollo
-            .mutate({
-              mutation: require('@/graphql/Agent/delete-agent.gql'),
-              variables: {
-                agentId: agent.id
-              }
-            })
-            .then(() => {
-              this.cleanUpDialog = false
-              setTimeout(() => {
-                agent.isDeleting = false
-              }, 10000)
-            })
-          setTimeout(() => {
-            agent.isDeleting = false
-          }, 2000)
+      this.clearingAgents = true
+      const unhealthyAgents = this.agents.filter(
+        agent => agent.secondsSinceLastQuery > 60 * this.unhealthyThreshold
+      )
+      if (!unhealthyAgents) {
+        LogRocket.captureMessage('Clean Up button open but no agents found')
+        this.setAlert({
+          alertShow: true,
+          alertMessage: 'Error clearing agents',
+          alertType: 'error'
         })
+      }
+      unhealthyAgents.forEach(agent => {
+        agent.isDeleting = true
+        this.$apollo
+          .mutate({
+            mutation: require('@/graphql/Agent/delete-agent.gql'),
+            variables: {
+              agentId: agent.ids
+            }
+          })
+          .then(() => {
+            this.cleanUpDialog = false
+            setTimeout(() => {
+              agent.isDeleting = false
+            }, 10000)
+          })
+          .catch(e => {
+            LogRocket.captureError(e)
+            setTimeout(() => {
+              agent.isDeleting = false
+              this.setAlert({
+                alertShow: true,
+                alertMessage: 'Error clearing agents',
+                alertType: 'error'
```

Comment:
Won't this attempt to show an alert for every agent? That could get messy in some scenarios. Maybe add a counter for failed mutations that we can show in a single alert at the end? (and maybe can be included in the logrocket error)

Reply:
Good point - can update that